internal/locate/region_cache.go

```diff
@@ -1178,8 +1178,13 @@ func (c *RegionCache) scheduleReloadRegion(region *Region) {
 	}
 	regionID := region.GetID()
 	if regionID > 0 {
+		const maxPendingReloadRegions = 512
 		c.regionsNeedReload.Lock()
-		c.regionsNeedReload.regions = append(c.regionsNeedReload.regions, regionID)
+		if len(c.regionsNeedReload.regions) < maxPendingReloadRegions {
+			c.regionsNeedReload.regions = append(c.regionsNeedReload.regions, regionID)
+		} else {
+			c.regionsNeedReload.regions[rand.Intn(maxPendingReloadRegions)] = regionID
```

Comment:
For this strategy, there seem to be starvation possibilities if `scheduleReloadRegion` keeps happening and no one can make progress.

Reply:
And the region whose reload is canceled should be set to non-async status(set `region.asyncReload` to 0), unless it'll not be reloaded in the future.

Reply:
Why not just ignore in this situation? Random indicates unpredictable...